### ISSUE TITLE
Migrate block-beam substraction statements from gpsans.api.process_single_configuration into its own function

### ISSUE THREAD

**Issue by glass-ships** [neutrons/drtsans#571]
Blocked beam subtraction is carried out within `gpsans.api.process_single_configuration()` as a sequence of statements.

**To-Do:**
- [ ] encapsulate the required statements for the subtraction of blocked-beam background into its own `subtract_blocked_beam()` function.

**Notes:**
It's OK that corrections to the blocked-beam run are same as those of the sample, namely:
- normalization by either time or monitor counts, depending of what the user required during reduction
- correction by solid angle
- correction by sensitivity
In summary, it's OK to apply `prepared_data()` to the blocked-beam with the configuration that is applied to the sample

**Comment by glass-ships** [comment]
This issue was migrated from [code.ornl.gov](https://code.ornl.gov/sns-hfir-scse/sans/sans-backend/-/issues/525).
